lib/addressable/idna/pure.rb

```diff
@@ -77,7 +79,7 @@ def self.to_ascii(input)
             part.force_encoding(Encoding::ASCII_8BIT)
           end
           if part =~ UTF8_REGEX && part =~ UTF8_REGEX_MULTIBYTE
-            ACE_PREFIX + punycode_encode(unicode_normalize_kc(part))
+            ACE_PREFIX + punycode_encode(part)
```

Comment:
And we need to do the normalise while we have an unicode string so before the conversion to `ASCII_8BIT`
Also the normalize should be before the split according to http://www.unicode.org/reports/tr46/#ProcessingStepNormalize.
So in the end it's both simpler and more correct this way :)